compiler/rustc_mir/src/transform/check_unsafety.rs

```diff
@@ -232,37 +232,45 @@ impl<'a, 'tcx> Visitor<'tcx> for UnsafetyChecker<'a, 'tcx> {
                     UnsafetyViolationKind::GeneralAndConstFn,
                     UnsafetyViolationDetails::DerefOfRawPointer,
                 ),
-                ty::Adt(adt, _) => {
-                    if adt.is_union() {
-                        if context == PlaceContext::MutatingUse(MutatingUseContext::Store)
-                            || context == PlaceContext::MutatingUse(MutatingUseContext::Drop)
-                            || context == PlaceContext::MutatingUse(MutatingUseContext::AsmOutput)
-                        {
-                            let elem_ty = match elem {
-                                ProjectionElem::Field(_, ty) => ty,
-                                _ => span_bug!(
-                                    self.source_info.span,
-                                    "non-field projection {:?} from union?",
-                                    place
-                                ),
-                            };
-                            if !elem_ty.is_copy_modulo_regions(
+                ty::Adt(adt, _) if adt.is_union() => {
+                    let assign_to_field = context
+                        == PlaceContext::MutatingUse(MutatingUseContext::Store)
```

Comment:
This strikes me as rather suboptimal formatting, bur rustfmt insists on it...

Reply:
Ouch.  This is poor even in isolation, but with the `||` chain it's terrible 🙁

You're in the compiler, so I guess you could try `matches!(context, PlaceContext::MutatingUse(MutatingUseContext::Store | MutatingUseContext::Drop | MutatingUseContext::AsmOutput)`?

Reply:
I reported this against rustfmt: https://github.com/rust-lang/rustfmt/issues/4492

Reply:
Your proposal is formatted slightly better:
```rust
let assign_to_field = matches!(
                        context,
                        PlaceContext::MutatingUse(
                            MutatingUseContext::Store
                                | MutatingUseContext::Drop
                                | MutatingUseContext::AsmOutput
                        )
                    );
```
Still not great, but the three alternatives do not fit on one line so it is not easy to make this nice.